File: backend/app/schemas/risk.py
from datetime import date, datetime, time
from pydantic import BaseModel
from typing import Optional


class RiskEventBase(BaseModel):
    event_type: str
    event_level: str = "general"
    event_date: date
    event_time: time
    location: Optional[str] = None
    description: str
    causes: Optional[str] = None
    injuries: Optional[str] = None
    immediate_measures: Optional[str] = None
    witnesses: Optional[str] = None
    status: str = "reported"
    handling_result: Optional[str] = None
    follow_up_plan: Optional[str] = None
    remark: Optional[str] = None


class RiskEventCreate(RiskEventBase):
    elder_id: int


class RiskEventUpdate(BaseModel):
    event_type: Optional[str] = None
    event_level: Optional[str] = None
    event_date: Optional[date] = None
    event_time: Optional[time] = None
    location: Optional[str] = None
    description: Optional[str] = None
    causes: Optional[str] = None
    injuries: Optional[str] = None
    immediate_measures: Optional[str] = None
    witnesses: Optional[str] = None
    status: Optional[str] = None
    handling_result: Optional[str] = None
    follow_up_plan: Optional[str] = None
    remark: Optional[str] = None


class RiskEventResponse(RiskEventBase):
    id: int
    elder_id: int
    reported_by_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/api/risk.ts
import api from './client'

export interface RiskEvent {
  id: number
  elder_id: number
  event_type: string
  event_level: string
  event_date: string
  event_time: string
  location?: string
  description: string
  causes?: string
  injuries?: string
  immediate_measures?: string
  reported_by_id?: number
  witnesses?: string
  status: string
  handling_result?: string
  follow_up_plan?: string
  remark?: string
  created_at: string
  updated_at: string
}

export const riskApi = {
  getList: (params?: {
    page?: number
    page_size?: number
    event_type?: string
    event_level?: string
    status?: string
    elder_id?: number
  }) => {
    return api.get<any>('/risk-events', { params })
  },

  getDetail: (id: number) => {
    return api.get<RiskEvent>(`/risk-events/${id}`)
  },

  create: (data: Partial<RiskEvent> & { elder_id: number }) => {
    return api.post<RiskEvent>('/risk-events', data)
  },

  update: (id: number, data: Partial<RiskEvent>) => {
    return api.put<RiskEvent>(`/risk-events/${id}`, data)
  },

  updateStatus: (id: number, status: string) => {
    return api.patch<RiskEvent>(`/risk-events/${id}/status`, { status })
  },

  remove: (id: number) => {
    return api.delete(`/risk-events/${id}`)
  },

  createIncident: (riskEventId: number) => {
    return api.post<any>(`/risk-events/${riskEventId}/create-incident`)
  },
}
